ledger/sandbox-classic/src/main/scala/platform/sandbox/SandboxServer.scala

```diff
@@ -70,7 +70,15 @@ object SandboxServer {
 
   // We memoize the engine between resets so we avoid the expensive
   // repeated validation of the sames packages after each reset
-  private val engine = new Engine(engineConfig)
+  private[this] var engine: Option[Engine] = None
+
+  private def getEngine(config: EngineConfig): Engine = synchronized {
+    engine.getOrElse {
+      val engine = new Engine(config)
```

Comment:
So if this is called twice with different configs, it will just return an engine configured wrongly?

As the `config` will always be the same, perhaps just store it in a variable outside?

Reply:
Let's solve this issue in  #7127